src/Aspire.Hosting.Azure.ContainerApps/AzureContanierAppsInfrastructure.cs

```diff
@@ -0,0 +1,991 @@
+        }
+
+        private async Task<ContainerAppContext> ProcessResourceAsync(IResource resource, DistributedApplicationExecutionContext executionContext, CancellationToken cancellationToken)
+        {
+            if (!_containerApps.TryGetValue(resource, out var context))
+            {
+                _containerApps[resource] = context = new ContainerAppContext(resource, this);
+                await context.ProcessResourceAsync(executionContext, cancellationToken).ConfigureAwait(false);
+            }
+
+            return context;
+        }
+
+        private sealed class ContainerAppContext(IResource resource, ContainerAppEnviromentContext containerAppEnviromentContext)
+        {
+            private readonly Dictionary<object, string> _allocatedParameters = [];
+            private readonly Dictionary<string, BicepParameter> _bicepParameters = [];
+            private readonly ContainerAppEnviromentContext _containerAppEnviromentContext = containerAppEnviromentContext;
+
+            record struct EndpointMapping(string Scheme, string Host, int Port, int? TargetPort, bool IsHttpIngress, bool External);
+
+            private readonly Dictionary<string, EndpointMapping> _endpointMapping = [];
+
+            private (int? Port, bool Http2, bool External)? _httpIngress;
+            private readonly List<int> _additionalPorts = [];
+
+            private BicepParameter? _managedIdentityIdParameter;
+            private BicepParameter? _containerRegistryUrlParameter;
+            private BicepParameter? _containerRegistryManagedIdentityIdParameter;
+
+            public IResource Resource => resource;
+
+            // Set the parameters to add to the bicep file
+            public Dictionary<string, IManifestExpressionProvider> Parameters { get; } = [];
+
+            public List<ContainerAppEnvironmentVariable> EnvironmentVariables { get; } = [];
+
+            public List<ContainerAppWritableSecret> Secrets { get; } = [];
+
+            public List<BicepValue<string>> Args { get; } = [];
+
+            public List<(ContainerAppVolume, ContainerAppVolumeMount)> Volumes { get; } = [];
+
+            public void BuildContainerApp(ResourceModuleConstruct c)
+            {
+                var containerAppIdParam = AllocateParameter(_containerAppEnviromentContext.ContainerAppEnvironmentId);
+
+                BicepParameter? containerImageParam = null;
+
+                if (!resource.TryGetContainerImageName(out var containerImageName))
+                {
+                    AllocateContainerRegistryParameters();
+
+                    containerImageParam = AllocateContainerImageParameter();
+                }
+
+                var containerAppResource = new ContainerApp(resource.Name, "2024-03-01")
+                {
+                    Name = resource.Name.ToLowerInvariant()
+                };
+
+                c.Add(containerAppResource);
+
+                // TODO: Add managed identities only when required
```

Comment:
What exactly does this mean? Can we elaborate a bit more (or open an issue on it)?

Reply:
My old code had logic to detect references to azure dependencies with managed identity and only generate this in that case.

Reply:
Can you expand this comment or log an issue for this?